### PR TITLE
Allow CAs with intermediate authorities.

### PR DESCRIPTION
**Pull request by matthughes**
Same PR as https://github.com/elasticsearch/logstash-forwarder/pull/261 but written for Log Courier.

I'm not sure how all the gemspec/version changes happened.  I checked out a clean branch from 'develop' and made a change to one file.  Then 'make' modified those files.  If that's not desired, please instruct on what I need to do.

### CODE REVIEW

**Comment by driskell** [comment]
Just leave out the version changes. If you commit to revert them and then debase interactive to squash to one commit. Or just reset to develop and apply changes again. Might need to force push either way.

Reason was to help work out if someone was running a released version or the develop branch. Since you built from develop it updated version to include commit so it's easier for someone to locate what code was used to build.

Thanks for the contribution. I'll review asap.

File: src/lc-lib/transports/tcp.go [hunk 1]
```diff
@@ -105,26 +104,29 @@ func NewTcpTransportFactory(config *core.Config, config_path string, unused map[
...
+      var pemBlockNum = 1
+      for {
+        block, rest = pem.Decode(rest)
+        if block != nil {
+          if block.Type != "CERTIFICATE" {
+            return nil, fmt.Errorf("Block %d does not contain a certificate: %s\n", pemBlockNum, ret.SSLCA)
+          }
+          cert, err := x509.ParseCertificate(block.Bytes)
+          if err != nil {
+            return nil, fmt.Errorf("Failed to parse CA certificate in block %d: %s\n", ret.SSLCA, pemBlockNum)
```
Comment (driskell) [inline src/lc-lib/transports/tcp.go]: Are the parameters the wrong way around here?

Comment (matthughes) [inline src/lc-lib/transports/tcp.go]: Yup, good find.

**Comment by matthughes** [comment]
Pushed a change stripping the versions and addressing your comment.

**Comment by driskell** [comment]
LGTM

**Pull request merged**